pkgs/development/compilers/open-watcom-bin/default.nix

```diff
@@ -0,0 +1,59 @@
+{ stdenvNoCC, fetchurl, unzip, runCommand, writeScript, file }:
+
+let
+  version = "1.9";
+
+  # The installer is a statically linked ELF binary with a ZIP file in it.
+  installerBin = fetchurl {
+    url = "ftp://ftp.openwatcom.org/install/open-watcom-c-linux-${version}";
+    sha256 = "1wzkvc6ija0cjj5mcyjng5b7hnnc5axidz030c0jh05pgvi4nj7p";
+  };
+
+  # Convert the self-extracting ZIP file into a tarball.
+  compilerTar = runCommand "open-watcom-c-linux-${version}.tar.xz" {} ''
+    mkdir open-watcom-${version}
+    ${unzip}/bin/unzip ${installerBin} -d open-watcom-${version}
+
+    tar -cJf $out open-watcom-${version}
+  '';
+
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "open-watcom-bin";
+  inherit version;
+
+  nativeBuildInputs = [ file ];
+
+  src = compilerTar;
```

Comment:
you should be able to just set `unpackCmd` to teach nix how to unpack the self-extracting zipfile.

Reply:
Fair enough. I've changed the whole scheme to operate the installer via `expect` instead. This gets rid of manually fixing up permissions.

Reply:
Oh wow, this is both nice and mad. Thanks :-D